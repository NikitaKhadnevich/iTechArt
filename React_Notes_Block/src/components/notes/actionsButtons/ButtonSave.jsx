import React from 'react';
import PropTypes from 'prop-types';
import IconButton from '@mui/material/IconButton';
import Stack from '@mui/material/Stack';
import SaveIcon from '@mui/icons-material/Save';

const ButtonSave = ({ id, saveEditedNote, updateDescription }) => (
  <Stack direction='row' spacing={0.5} alignItems='center'>
    <IconButton
      id={id}
      aria-label='save'
      color='primary'
      onClick={() => saveEditedNote(id, updateDescription)}
    >
      <SaveIcon />
    </IconButton>
  </Stack>
);

ButtonSave.propTypes = {
  saveEditedNote: PropTypes.func,
  updateDescription: PropTypes.string,
  id: PropTypes.number,
};
ButtonSave.defaultProps = {
  saveEditedNote: 'saveEditedNote',
  updateDescription: 'updateDescription',
  id: 'id',
};

export default ButtonSave;
