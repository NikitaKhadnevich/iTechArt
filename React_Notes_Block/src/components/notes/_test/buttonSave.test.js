/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ButtonSave from '../actionsButtons/ButtonSave';

configure({ adapter: new Adapter() }); // ES6 modules

describe('ButtonSave', () => {
  const buttonSaveProps = {
    id: 1,
    handleSaveNote: jest.fn(
      (id, updateDescription) =>
        `call save with index: ${id} and data: ${updateDescription}`
    ),
    updateDescription: [
      {
        id: '',
        title: 'New Title',
        description: 'Some description',
        date: '10/22/2021, 04:10:12 PM',
        isActive: false,
        isChange: false,
        isShared: false,
        sharedTime: '',
      },
    ],
  };

  it('Is it ButtonSave Exist', () => {
    const buttonSave = shallow(<ButtonSave {...buttonSaveProps} />);
    expect(buttonSave).toMatchSnapshot();
    expect(buttonSave).toHaveLength(1);
  });
  it('Test Save Event', () => {
    const buttonSave = shallow(<ButtonSave {...buttonSaveProps} />);
    const { id, handleSaveNote, updateDescription } = buttonSaveProps;
    buttonSave.find('ForwardRef(IconButton)').simulate('click');
    expect(handleSaveNote).toHaveBeenCalledTimes(1);
    expect(handleSaveNote).toHaveBeenCalledWith(id, updateDescription);
  });
});
