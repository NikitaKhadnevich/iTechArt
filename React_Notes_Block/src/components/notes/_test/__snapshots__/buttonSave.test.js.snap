// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSave Is it ButtonSave Exist 1`] = `
<ForwardRef(Stack)
  alignItems="center"
  direction="row"
  spacing={0.5}
>
  <ForwardRef(IconButton)
    aria-label="save"
    color="primary"
    id={1}
    onClick={[Function]}
  >
    <Memo(ForwardRef(SaveIcon)) />
  </ForwardRef(IconButton)>
</ForwardRef(Stack)>
`;
